discovery-provider/es-indexer/etl/playlistEtl.ts

```diff
@@ -0,0 +1,154 @@
+import _ from 'lodash'
+import { keyBy } from 'lodash'
+import { PlaylistDoc } from '../types/docs'
+import { dialPg } from './conn'
+import { indexNames } from './indexNames'
+import { BlocknumberCheckpoint, Job } from './job'
+
+/*
+don't know how to do the sql to populate moods, genres, tags
+based on join with playlist_contents
+
+so uses withBatch callback to load that stuff up and attach it
+*/
+
+export const playlistEtl: Job = {
+  tableName: 'playlists',
+  idField: 'playlist_id',
+  indexBatchSize: 4000,
```

Comment:
reason for 4K batch size?

Reply:
elasticsearch has a number of "circuit breaker" settings, so there are limits on bulk index request like number of documents, total size of bulk request.

Our tracks records are much larger (in bytes) than playlists... so that has a smaller batch size.  But playlists are smaller so we can do more per-batch.

This [go library](https://github.com/olivere/elastic/wiki/BulkIndex) for instance has a fancy bulk processor that measures the total size of a request and dispatches at a threshold which is a more scientific way to do it.  But the layperson approach is to pick a "safe" number considering the documents.